File: src/sub/java/yuuto/enhancedinventories/inventory/InventoryMerged.java
package yuuto.enhancedinventories.inventory;

import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.inventory.IInventory;
import net.minecraft.item.ItemStack;

public class InventoryMerged extends InventoryBase{
	
	ItemStack[][] inv;
	int totalLength;
	int tempSlot;
	public InventoryMerged(ItemStack[]... stacks){
		this.inv = stacks;
		for(ItemStack[] s : stacks){
			totalLength += s.length;
		}
	}
	@Override
	public int getSizeInventory() {
		return totalLength;
	}
	public ItemStack[] getInvAndSlot(int slot){
		tempSlot = slot;
		for(int i = 0; i < inv.length; i++){
			if(slot < inv[i].length)
				return inv[i];
			tempSlot -= inv[i].length;
		}
		return null;
	}
	@Override
	public ItemStack getStackInSlot(int p_70301_1_) {
		return getInvAndSlot(p_70301_1_)[tempSlot];
	}
	@Override
	public void setInventorySlotContents(int p_70299_1_, ItemStack p_70299_2_) {
		getInvAndSlot(p_70299_1_)[tempSlot] = p_70299_2_;
	}
	@Override
	public String getInventoryName() {
		// TODO Auto-generated method stub
		return null;
	}
	@Override
	public boolean hasCustomInventoryName() {
		// TODO Auto-generated method stub
		return false;
	}
	@Override
	public void markDirty() {
		// TODO Auto-generated method stub
		
	}
	@Override
	public boolean isUseableByPlayer(EntityPlayer p_70300_1_) {
		// TODO Auto-generated method stub
		return false;
	}
	@Override
	public void openInventory() {
		// TODO Auto-generated method stub
		
	}
	@Override
	public void closeInventory() {
		// TODO Auto-generated method stub
		
	}
	
	

}

File: src/main/scala/yuuto/enhancedinventories/inventory/InventoryMerged.scala
package yuuto.enhancedinventories.inventory

import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.item.ItemStack;
import yuuto.yuutolib.inventory.InventoryBase

class InventoryMerged(inv:Seq[Array[ItemStack]]) extends InventoryBase{
	protected var totalLength:Int=0;
  protected var tempSlot:Int=0;
  for(s <-inv){
    totalLength+=s.length;
  }
	/*public InventoryMerged(ItemStack[]... stacks){
		this.inv = stacks;
		for(ItemStack[] s : stacks){
			totalLength += s.length;
		}
	}*/
	override def getSizeInventory():Int=totalLength;
	def getInvAndSlot(slot:Int):Array[ItemStack]={
		tempSlot = slot;
		for(i <- 0 until inv.length){
			if(tempSlot < inv.apply(i).length){
				return inv.apply(i);
			}
			tempSlot -= inv.apply(i).length;
		}
		return null;
	}
	override def getStackInSlot(slot:Int):ItemStack=getInvAndSlot(slot)(tempSlot);
	override def setInventorySlotContents(slot:Int , stack:ItemStack)=getInvAndSlot(slot)(tempSlot) = stack;
	override def getInventoryName():String=null;
	override def hasCustomInventoryName():Boolean=false;
	override def markDirty(){}
	override def isUseableByPlayer(player:EntityPlayer):Boolean=false;
	override def openInventory(){}
	override def closeInventory(){}
	
	

}